piperider_cli/dbtutil.py

```diff
@@ -350,6 +351,245 @@ def is_chosen(key, metric):
+                return None
+
+            return m
+        elif metric.get('type') == 'derived':
+            ref_metrics = []
+            time_grains = ['day', 'month', 'year']
+            expr = metric.get('type_params').get('expr')
+            for ref_metric in metric.get('type_params').get('metrics'):
+                if ref_metric.get('offset_window') is not None:
+                    console.print(
+                        f"[[bold yellow]Skip[/bold yellow]] Metric '{metric.get('name')}'. "
+                        f"Derived metric property 'offset_window' is not supported.")
+                    statistics.add_field_one('nosupport')
+                    return None
+                m2 = _create_metric(
+                    ref_metric.get('name'),
+                    filter=ref_metric.get('filter'),
+                    alias=ref_metric.get('alias'))
+                ref_metrics.append(m2)
+
+                derived_time_grains = find_derived_time_grains(manifest, metric_map[ref_metric.get('name')])
+                if len(time_grains) < len(derived_time_grains):
+                    time_grains = derived_time_grains
+
+                if ref_metric.get('alias') is not None:
+                    expr = expr.replace(ref_metric.get('alias'), ref_metric.get('name'))
+
+            m = Metric(metric.get('name'),
+                       calculation_method='derived',
+                       expression=expr,
+                       time_grains=time_grains,
+                       label=metric.get('label'), description=metric.get('description'), ref_metrics=ref_metrics,
+                       ref_id=metric.get('unique_id'))
+            return m
+        elif metric.get('type') == 'ratio':
+            ref_metrics = []
+            time_grains = ['day', 'month', 'year']
+            numerator = metric.get('type_params').get('numerator')
+            m2 = _create_metric(
+                numerator.get('name'),
+                filter=numerator.get('filter'),
+                alias=numerator.get('alias'))
+            ref_metrics.append(m2)
+            derived_time_grains = find_derived_time_grains(manifest, metric_map[numerator.get('name')])
+            if len(time_grains) < len(derived_time_grains):
+                time_grains = derived_time_grains
+
+            denominator = metric.get('type_params').get('denominator')
+            m2 = _create_metric(
+                denominator.get('name'),
+                filter=denominator.get('filter'),
+                alias=denominator.get('alias'))
+            ref_metrics.append(m2)
+            derived_time_grains = find_derived_time_grains(manifest, metric_map[denominator.get('name')])
+            if len(time_grains) < len(derived_time_grains):
+                time_grains = derived_time_grains
+
+            m = Metric(metric.get('name'),
+                       calculation_method='derived',
+                       expression=f"{numerator.get('name')} / {denominator.get('name')}",
```

Comment:
nitpick: Would be better to pass the calculation_method & numerator & denominator to Metric object. Then generate the SQL accordingly. But now, it is fine.

Reply:
Will put it into another story to enhance